packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -54,31 +83,64 @@ export function CrossmintWalletProvider({
             return { startedCreation: false, reason: `Jwt not set in "CrossmintProvider".` };
         }
 
-        const internalCall = async () => {
-            try {
-                setState({ status: "in-progress" });
-                const wallet = await smartWalletSDK.getOrCreateWallet(
-                    { jwt: crossmint.jwt as string },
-                    defaultChain,
-                    config
-                );
-                setState({ status: "loaded", wallet });
-            } catch (error: unknown) {
-                console.error("There was an error creating a wallet ", error);
-                setState(deriveErrorState(error));
-            }
-        };
-
-        internalCall();
+        try {
+            setWalletState({ status: "in-progress" });
+            const wallet = await smartWalletSDK.getOrCreateWallet(
+                { jwt: crossmint.jwt as string },
+                defaultChain,
+                enhanceConfigWithPasskeyPrompts(config)
+            );
+            setWalletState({ status: "loaded", wallet });
+        } catch (error: unknown) {
+            console.error("There was an error creating a wallet ", error);
+            setWalletState(deriveErrorState(error));
+        }
         return { startedCreation: true };
     };
 
+    const enhanceConfigWithPasskeyPrompts = (config: WalletConfig) => {
+        if (showWalletModals && (config.signer as PasskeySigner).type === "PASSKEY") {
+            return {
+                ...config,
+                signer: {
+                    ...config.signer,
+                    onPrePasskeyRegistration: createPasskeyPrompt("create-wallet"),
+                    onPasskeyRegistrationError: createPasskeyPrompt("create-wallet-error"),
+                    onFirstTimePasskeySigning: createPasskeyPrompt("transaction"),
+                    onFirstTimePasskeySigningError: createPasskeyPrompt("transaction-error"),
+                },
+            };
+        }
+        return config;
+    };
+
+    const createPasskeyPrompt = (type: ValidPasskeyPromptType) => () =>
+        new Promise<void>((resolve) => {
```

Comment:
do we need to wrap into a promise?
I thought non-async functions can always be used in a slot marked w/ `async`, just not the other way around